java/src/jmri/jmrix/can/cbus/CbusSensor.java

```diff
@@ -82,40 +82,32 @@ public void requestUpdateFromLayout() {
     }
 
     /**
-     * User request to set the state, which means that we broadcast that to all
-     * listeners by putting it out on CBUS. In turn, the code in this class
+     * User request to set the state.
+     * We broadcast that to all listeners by putting it out on CBUS. 
+     * In turn, the code in this class
      * should use setOwnState to handle internal sets and bean notifies.
-     * Unknown state does not send a message to CBUS but updates 
-     * internal sensor state, enabling user test of Start of Day / Logix.
+     * Unknown / Inconsistent states do not send a message to CBUS,
+     * but do update sensor state.
      * {@inheritDoc}
      */
     @Override
     public void setKnownState(int s) throws jmri.JmriException {
+        setOwnState(s);
         CanMessage m;
-        if (s == Sensor.ACTIVE) {
-            if (getInverted()){
-                m = addrInactive.makeMessage(tc.getCanid());
-                setOwnState(Sensor.ACTIVE);
-            } else {
-                m = addrActive.makeMessage(tc.getCanid());
-                setOwnState(Sensor.ACTIVE);
-            }
-            CbusMessage.setPri(m, CbusConstants.DEFAULT_DYNAMIC_PRIORITY * 4 + CbusConstants.DEFAULT_MINOR_PRIORITY);
-            tc.sendCanMessage(m, this);
-        } else if (s == Sensor.INACTIVE) {
-            if (getInverted()){
-                m = addrActive.makeMessage(tc.getCanid());
-                setOwnState(Sensor.INACTIVE);                
-            } else {
-                m = addrInactive.makeMessage(tc.getCanid());
-                setOwnState(Sensor.INACTIVE);
-            }
-            CbusMessage.setPri(m, CbusConstants.DEFAULT_DYNAMIC_PRIORITY * 4 + CbusConstants.DEFAULT_MINOR_PRIORITY);
-            tc.sendCanMessage(m, this);
-        }
-        if (s == Sensor.UNKNOWN){
-            setOwnState(Sensor.UNKNOWN);
+        switch (s) {
+            case Sensor.ACTIVE:
+                m = ( getInverted() ? addrInactive.makeMessage(tc.getCanid()) : 
+                    addrActive.makeMessage(tc.getCanid()));
+                break;
+            case Sensor.INACTIVE:
+                m = ( !getInverted() ? addrInactive.makeMessage(tc.getCanid()) : 
+                    addrActive.makeMessage(tc.getCanid()));
+                break;
+            default:
+                return;
         }
+        CbusMessage.setPri(m, CbusConstants.DEFAULT_DYNAMIC_PRIORITY * 4 + CbusConstants.DEFAULT_MINOR_PRIORITY);
+        tc.sendCanMessage(m, this);
```

Comment:
One thing I didn't thought of before is that the message should only be sent to the CAN bus if the state goes to ACTIVE or INACTIVE. With this code, you call `tc.sendCanMessage(m, this)` with a null message if the state goes to UNKNOWN.

Please move the `CbusMessage.setPri(...)` and `tc.sendCanMessage(m, this)` calls to inside the switch case statement.

Reply:
The default in the switch is a return so no message sent, test added to confirm this.